### ISSUE TITLE
Citekey generation is broken with conditional formatting after v1.6.50

### ISSUE THREAD

**Issue by kaymmm** [retorquere/zotero-better-bibtex#499]
(new issue related to #478)

tl;dr: citekey generation using [>0] formatting is broken since 1.6.51 (or maybe .52). Works in 1.6.50.

There's something a bit wonky going on recently with the generation and display of citation keys that may or may not be related to #497, but I'm pretty sure it's related to the [>0] citation format which is why I'm adding to this issue.

In general, it seems that citation keys are not being auto-generated, and right-clicking to generate a citation key does nothing. At least, that's what I think. They're definitely not being displayed in the Extra or Citation Key fields.

The reason I can't quite figure things out is that I just tried going back through previous releases one-by-one until I hit 1.6.50. The issue persisted through all of the recent versions until I went back to 1.6.50, when it seemed like the auto-generation of citation keys was working correctly.

When I checked the .bib file that is being automatically sync'd (tried with all methods of manual/change/idle), entries for newly added items are not being generated unless the citekey generates correctly. Everything seems to be working correctly with 1.6.50.

I'm using the citekey format: [auth][>0][year]|[veryshorttitle][year]. This had been working since the [>0] format option was merged a month ago. Using [zotero] as the format seems to be working as expected. If I just use something like [auth][year] it works as expected, but obviously not if the author field is empty.

I submitted an error report, report ID: M62BNEU3

When I try to generate a citation key, the debug error that's generated is:

JavaScript Error: "TypeError: longest.replace is not a function"
If I manually add a bibtex: SomeKey to the "extra" field, it will sometimes stick, but even that has been problematic as the "citation key" field will remain blank until I change the "extra" field a couple of times, and sometimes requires me to restart Zotero.

I'm having the same problem on both standalone and zotero-ffx.

**Comment by retorquere** [comment]
Can you try https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-1.6.56-circle-2464.xpi ?

**Comment by kaymmm** [comment]
That seems to have fixed the problem! Thank you for working so quickly, you are amazing!

**Comment by retorquere** [comment]
If I was amazing, I really shouldn't be making these simple mistakes. Sorry
about that.
